api/src/main/java/org/openmrs/module/commonreports/reports/DRCCD4MonitoringReportManager.java

```diff
@@ -0,0 +1,281 @@
+		return true;
+		
+	}
+	
+	@Override
+	public String getVersion() {
+		return "1.0.0-SNAPSHOT";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "45631da8-1e16-414f-9894-4aafad370c77";
+	}
+	
+	@Override
+	public String getName() {
+		return MessageUtil.translate("commonreports.report.drc.cd4Monitoring.reportName");
+	}
+	
+	@Override
+	public String getDescription() {
+		return MessageUtil.translate("commonreports.report.drc.cd4Monitoring.reportDescription");
+	}
+	
+	private Parameter getStartDateParameter() {
+		return new Parameter("startDate", MessageUtil.translate("commonreports.report.util.reportingStartDate"), Date.class);
+	}
+	
+	private Parameter getEndDateParameter() {
+		return new Parameter("endDate", MessageUtil.translate("commonreports.report.util.reportingEndDate"), Date.class);
+	}
+	
+	public static String col1 = "";
+	
+	public static String col2 = "";
+	
+	public static String col3 = "";
+	
+	public static String col4 = "";
+	
+	public static String col5 = "";
+	
+	public static String col6 = "";
+	
+	public static String col7 = "";
+	
+	public static String col8 = "";
+	
+	public static String col9 = "";
+	
+	public static String col10 = "";
+	
+	public static String col11 = "";
+	
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> params = new ArrayList<Parameter>();
+		params.add(getStartDateParameter());
+		params.add(getEndDateParameter());
+		return params;
+	}
+	
+	@Override
+	public ReportDefinition constructReportDefinition() {
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		
+		// cd4Monitoring Grouping
+		CohortCrossTabDataSetDefinition cd4Monitoring = new CohortCrossTabDataSetDefinition();
+		cd4Monitoring.addParameters(getParameters());
+		rd.addDataSetDefinition(getName(), Mapped.mapStraightThrough(cd4Monitoring));
+		
+		Map<String, Object> parameterMappings = new HashMap<String, Object>();
+		parameterMappings.put("onOrAfter", "${startDate}");
+		parameterMappings.put("onOrBefore", "${endDate}");
+		SqlCohortDefinition sqd = new SqlCohortDefinition();
+		
+		// CD4 value in range
+		// Enrolled for HIV care
+		String sql = "SELECT DISTINCT p.patient_id FROM patient p WHERE p.voided = 0 "
+		        + "AND EXISTS (SELECT 1 FROM obs o_num JOIN concept c_num ON o_num.concept_id = c_num.concept_id WHERE o_num.person_id = p.patient_id AND c_num.uuid = '5497AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' AND o_num.voided = 0 AND o_num.value_numeric IS NOT NULL AND o_num.obs_datetime BETWEEN :onOrAfter AND :onOrBefore) "
+		        + "AND EXISTS (SELECT 1 FROM obs o_coded JOIN concept c_coded ON o_coded.concept_id = c_coded.concept_id WHERE o_coded.person_id = p.patient_id AND c_coded.uuid = '83e40f2c-c316-43e6-a12e-20a338100281' AND o_coded.voided = 0 AND o_coded.value_coded IS NOT NULL);";
```

Comment:
Shouldn't enrollment be covered by presence on a program?

Reply:
Moved this to rely on program.